lib/mix_templates.ex

```diff
@@ -615,11 +649,56 @@ end
       end
     end
 
+    defp generate_new_file(source, dest, assigns) do
+      case assigns[:assigns][:copied_files] do
+        :all ->
+          evaluate_eex(source, dest, assigns, true)
+        copied_files when is_map(copied_files) ->
+          file_extension = String.split(source, ".") |> List.last
```

Comment:
I thought about extracting the `copied_files when is_map(copied_files)` case into its own function, but I couldn't think of a good name for the function. Open to suggestions!